libs/components/src/lib/Accordion/index.tsx

```diff
@@ -0,0 +1,72 @@
+import cls from "classnames";
+import { FC, ReactNode, useState } from "react";
+import { PlusIcon } from "../Icons/plus";
+import classes from "./index.module.scss";
+import { useData } from "./index.hook";
+
+interface AccordionItemProps {
+  title: string;
+  children: ReactNode;
+}
+
+const AccordionItem: FC<AccordionItemProps> = ({ title, children }) => {
+  const {isOpen, handleToggle} = useData();
+
+  return (
+    <div className={classes.accordion__item} role="button" aria-expanded={isOpen}>
+      <div
+        className={cls(classes.accordion__item__header, {
+          [classes.accordion__item__header__open]: isOpen,
+        })}
+        onClick={handleToggle}
+        aria-controls={`accordion-item-${title}`}
+        aria-label={`Toggle ${title} accordion item`}
+        role="heading"
+        aria-level={3}
+      >
+        <span
+          className={cls(classes.accordion__item__header__title, {
+            [classes.accordion__item__header__title__open]: isOpen,
+          })}
+          role="contentinfo"
+        >
+          {title}
+        </span>
+        <div
+          className={cls(classes.accordion__item__header__icon, {
+            [classes.accordion__item__header__icon__open]: isOpen,
+          })}
+          id={`accordion-item-${title}`}
+        >
+          <PlusIcon />
+        </div>
+      </div>
+      <div
+        className={cls(classes.accordion__item__children, {
+          [classes.accordion__item__children__open]: isOpen,
+        })}
+        role="list"
+      >
+        {children}
+      </div>
+    </div>
+  );
+};
+
+interface AccordionProps {
+  items?: AccordionItemProps[];
+}
+
+const Accordion: FC<AccordionProps> = ({ items }) => {
```

Comment:
const Accordion to export const Accordion and remove export in line 72.

Reply:
### This needs more discussion. 
AccordionItem manages its own "state". Using a render method like renderAccordion item brings the state to the parent. Then for accessing "state" in renderAccordionItem we only will be able to access "state" like this:

```
const [state, useState] = useState([])
items.map((item, index) => renderAccordionItem(item[index])
```
Managing "state" in the AccordionItem component lowers unnecessary rerenders and any possible implications that using the index in the map function will bring. 

The Suggestion that creating two components is not accepted, I wanted to request to normalize more than one component in one module. 

[Reddit](https://www.reddit.com/r/reactjs/comments/uvhi7n/is_it_good_practice_to_put_multiple_components_in/): __is it good practice to put multiple components in one file like this?_

_I think that is absolutely fine. Think of it as a way of organizing the main component better by defining subcomponents in the same file; there's no benefit in adding cognitive overhead by moving these subcomponents to other files. I personally prefer a slightly larger file with a well-organized structure than having to jump across several files for components that are not being reused._